test.c

```diff
@@ -4316,6 +4316,9 @@ main (void)
   test_simple("GET / HTTP/11.1\r\n\r\n", HPE_INVALID_VERSION);
   test_simple("GET / HTTP/1.01\r\n\r\n", HPE_INVALID_VERSION);
 
+  test_simple("GET / HTTP/1.0\r\nHello: w\1rld\r\n\r\n", HPE_INVALID_HEADER_TOKEN);
```

Comment:
If you could also test with explicit `\0` I think it would be great, since NULs are particularly tricky when it comes to C strings as a special case.

Reply:
I agree that there's point to checking `\0` too. However, this works without further changes to the code, because `strlen()` still returns correct value. Perhaps, we should address it in a separate follow-up change?

Reply:
That's fine too.